### PR TITLE
Allow search customers by ID in admin customer ajax search

### LINKED ISSUE #13980: Customer search box does not search by ID
On new order screen.

### PR DESCRIPTION
**Pull request by claudiosanches**
This also remove the legacy `quietMillis` in favor of `delay`.

See:

Select2 3:
screenshot from 2017-04-06 01-54-49

Select2 4:
screenshot from 2017-04-06 01-54-58

I still prefer a command like `id:123`.

Closes #13980

### CODE REVIEW

File: includes/class-wc-ajax.php [hunk 1]
```diff
@@ -1229,8 +1229,26 @@ public static function json_search_customers() {
 			wp_die();
 		}
 
-		$data_store      = WC_Data_Store::load( 'customer' );
-		$ids             = $data_store->search_customers( $term );
+		// Stop if it is not numeric and smaller than 3 characters.
+		if ( ! is_numeric( $term ) && 2 >= strlen( $term ) ) {
```
Comment (mikejolley) [inline includes/class-wc-ajax.php]: If you have made the input accept 1 char, why is this needed? Just check numeric.

Comment (claudiosanches) [inline includes/class-wc-ajax.php]: Still checking for less than 3 characters it's a burden generating long queries.

Comment (mikejolley) [inline includes/class-wc-ajax.php]: It's not a burden for numeric queries. Just needs to see if ID exists?

Comment (claudiosanches) [inline includes/class-wc-ajax.php]: But this stop the query if isn't numeric... That's why `! is_numeric( $term )`.

Comment (mikejolley) [inline includes/class-wc-ajax.php]: Move along. It's fine :p

Comment (claudiosanches) [inline includes/class-wc-ajax.php]: Just waiting for a "Status: Approved" :ok_hand:
